drivers/adc-dac/ad5592r/ad5593r.c

```diff
@@ -121,6 +126,50 @@ int32_t ad5593r_read_adc(struct ad5592r_dev *dev, uint8_t chan,
 	return 0;
 }
 
+/**
+ * Read Multiple ADC Channels.
+ *
+ * @param dev - The device structure.
+ * @param chans - The ADC channels to be readback
+ * @param values - ADC value array
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t ad5593r_multi_read_adc(struct ad5592r_dev *dev, uint16_t chans,
+			       uint16_t *values)
+{
+	uint8_t data[AD5593R_ADC_VALUES_BUFF_SIZE], i;
+	int32_t ret;
+	uint8_t samples;
+
+	if (!dev)
+		return FAILURE;
```

Comment:
Same as above, either you use error codes through the entire function, either `return -1`